### PR TITLE
TSBM-23 Debug renesas platform to be able to join a network

### PR DESCRIPTION
**Pull request by NeethuVisw**

### CODE REVIEW

File: examples/lock-app/renesas/args.gni [hunk 1]
```diff
@@ -19,6 +19,8 @@ import("${chip_root}/src/platform/renesas/args.gni")
 
 renesas_sdk_target = get_label_info(":sdk", "label_no_toolchain")
 
+cflags = [ "-fno-exceptions" ]
```
Comment (NeethuVisw) [inline examples/lock-app/renesas/args.gni]: this is no longer needed

File: examples/lock-app/renesas/args.gni [hunk 2]
```diff
@@ -29,6 +29,7 @@ chip_progress_logging = true
 chip_detail_logging = true
 chip_automation_logging = false
 treat_warnings_as_errors = false
+use_exceptions = false
```
Comment (NeethuVisw) [inline examples/lock-app/renesas/args.gni]: no longer needed

File: src/platform/renesas/PlatformManagerImpl.cpp [hunk 3]
```diff
@@ -30,5 +42,29 @@ namespace DeviceLayer {
  */
 PlatformManagerImpl PlatformManagerImpl::sInstance;
 
+// CHIP_ERROR PlatformManagerImpl::_InitChipStack(void)
```
Comment (NeethuVisw) [inline src/platform/renesas/PlatformManagerImpl.cpp]: remove if not needed

File: src/app/server/Server.cpp [hunk 4]
```diff
@@ -75,7 +75,8 @@ class DeviceTypeResolver : public chip::Access::AccessControl::DeviceTypeResolve
 public:
     bool IsDeviceTypeOnEndpoint(chip::DeviceTypeId deviceType, chip::EndpointId endpoint) override
     {
-        return chip::app::IsDeviceTypeOnEndpoint(deviceType, endpoint);
+        //return chip::app::IsDeviceTypeOnEndpoint(deviceType, endpoint);
+        return false;
```
Comment (NeethuVisw) [inline src/app/server/Server.cpp]: verify return

File: src/platform/renesas/KeyValueStoreManagerImpl.cpp [hunk 5]
```diff
@@ -17,14 +17,84 @@
  */
 
 #include <platform/KeyValueStoreManager.h>
+#include <unordered_map>
```
Comment (NeethuVisw) [inline src/platform/renesas/KeyValueStoreManagerImpl.cpp]: include standard headers 1st

Comment (n-nievergeld) [reply src/platform/renesas/KeyValueStoreManagerImpl.cpp]: Depends: for zephyr(for example), zephyr/kernel always needs to be included first.

Also what we did at my previous employer, we did system includes last, just as a step for extra validation of dependencies(there might be more reasons).

File: src/include/platform/internal/GenericConfigurationManagerImpl.ipp [hunk 6]
```diff
@@ -356,7 +356,13 @@ CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::StoreSerialNumber(const
...
 CHIP_ERROR GenericConfigurationManagerImpl<ConfigClass>::GetPrimaryWiFiMACAddress(uint8_t * buf)
 {
-    return CHIP_DEVICE_ERROR_CONFIG_NOT_FOUND;
+    buf[0] = 0x11;
+    buf[1] = 0x22;
+    buf[2] = 0x33;
+    buf[3] = 0x44;
+    buf[4] = 0x55;
+    buf[5] = 0x66;
+    return CHIP_NO_ERROR;
```
Comment (n-nievergeld) [inline src/include/platform/internal/GenericConfigurationManagerImpl.ipp]: I restored this

File: src/platform/renesas/CHIPDevicePlatformConfig.h [hunk 7]
```diff
@@ -25,29 +25,14 @@
 #pragma once
 
 // -------------------- General Configuration --------------------
-#define CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE 0x00001300
+#define CHIP_DEVICE_CONFIG_CHIP_TASK_STACK_SIZE 0x00002000
```
Comment (n-nievergeld) [inline src/platform/renesas/CHIPDevicePlatformConfig.h]: I am trying to get it down to 0x1500 as I see that is the maximum usage 0x13F8 is the maximum usage right now.

Comment (n-nievergeld) [reply src/platform/renesas/CHIPDevicePlatformConfig.h]: Settled on 0x1600

**Comment by NeethuVisw** [comment]
it looks fine

**Review by n-nievergeld — APPROVED** [review]

**Pull request merged**